src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -242,5 +245,7 @@ public object CreateInstance(Type t)
         ConcurrentDictionary<RuntimeTypeHandle, RuntimeTypeHandle> concreteToInterfaceTypeMapping = new ConcurrentDictionary<RuntimeTypeHandle, RuntimeTypeHandle>();
         ConcurrentDictionary<string, RuntimeTypeHandle> nameToType = new ConcurrentDictionary<string, RuntimeTypeHandle>();
         ConcurrentDictionary<RuntimeTypeHandle, RuntimeMethodHandle> typeToConstructor = new ConcurrentDictionary<RuntimeTypeHandle, RuntimeMethodHandle>();
+
+        private static readonly object messageInitializationLock = new object();
```

Comment:
ah I actually wanted to remove the static. According to the tests, it works with having multiple instances since it seems to be able to have that dynamic assembly multiple times even with the same name.